llama_index/llms/openai_like.py

```diff
@@ -58,10 +71,19 @@ def _tokenizer(self) -> Tokenizer:
                     "huggingface tokenizers with OpenAILike."
                 ) from exc
 
-            return AutoTokenizer.from_pretrained(self._get_model_name())
```

Comment:
This wasn't quite correct, the model `OpenAILike` wraps here isn't a tokenizer. `_get_model_name` will 9/10 times return a non-tokenizer model.

We needed the `tokenizer` field to be flexible to this.

Reply:
Makes sense I think given the logic of _get_model_name checks for `ft` and splits the string, which I think is pretty narrow to OpenAI. What model names were you testing out of curiousity?

Reply:
With `LocalAI` or `llama-cpp-python`, you get access to many models. Check LocalAI's model gallery to get some ideas: https://localai.io/models/ --> https://github.com/go-skynet/model-gallery

For example, their Hugging Face configs are pretty extensive: https://raw.githubusercontent.com/go-skynet/model-gallery/main/huggingface.yaml